pkg/app/instances/docker.go

```diff
@@ -423,6 +428,15 @@ func (m *DockerInstanceManager) deleteDockerVolumeIfNeeded(ctx context.Context,
 	if len(listRes) > 0 {
 		return nil
 	}
+
+	mu := m.getRWMutex(user)
+	if locked := mu.TryLock(); !locked {
+		// If it can't acquire lock on this mutex, there's ongoing host
+		// creation with this volume or deletion of this volume. For these
+		// cases, it doesn't need to delete docker volume here.
+		return nil
```

Comment:
This would return success (nil) but leave the host around if the request arrives while it's still being created. A better approach would be lock the mutex (possibly waiting for the lock to be acquired) and delete the host if it still exists.